probe-rs/src/architecture/arm/sequences.rs

```diff
@@ -397,10 +397,9 @@ fn cortex_m_reset_system(interface: &mut dyn ArmMemoryInterface) -> Result<(), A
         if !dhcsr.s_reset_st() {
```

Comment:
As I've mentioned on Matrix, logic here seems inverted, doesn't it?

If anything we should read DHCSR right before writing to AIRCR (just to be sure we don't have some leftover sticky bits) and then have a `DHCSR.S_RESET_ST == 0b1` checking loop with a timeout.

Reply:
I am rather confused. How is ANYTHING working if this is wrong (which I agree on)?

Maybe it usually resets way faster than the second read so until we read again we already reset and it works because our first read reads 1, gets discarded and then the second reads 0 and exits the loop?

Reply:
Sounds good, let's PR it in a different PR :) I think master is fine to break stuff. Plus, we have quite a few HW runners :)